src/web/View.php

```diff
@@ -643,6 +649,48 @@ public function clearJsBuffer(bool $scriptTag = true)
     }
 
     /**
+     * Registers a generic <script> code block.
+     * @param string $script the generic <script> code block to be registered
+     *
+     * @param string $key the key that identifies the generic <script> code block. If null, it will use
+     * $script as the key. If two generic <script> code blocks are registered with the same key, the latter
+     * will overwrite the former.
+     */
+    public function registerScript($script, $options = [], $key = null)
+    {
+        $key = $key ?: md5($script);
+        $this->_scripts[$key] = Html::script($script, $options);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function renderHeadHtml()
+    {
+        $lines = [];
+        if (!empty($this->title)) {
+            $lines[] = '<title>' . Html::encode($this->title) . '</title>' . "\n";
```

Comment:
It's something else that I'm hoping can be added. I can do a separate PR if you like; it emulates what Yii2 does in their default template `yiisoft/yii2-debug/views/layouts/main.php`

If `$title` is not set (as is the case by default), then nothing happens. If it is set, then it renders the `<title>` tag based on `$view->title`